platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/ConfigurationCacheHost.kt

```diff
@@ -24,9 +25,9 @@ import java.io.File
 @ServiceScope(Build::class)
 interface ConfigurationCacheHost : HostServiceProvider {
 
-    val currentBuild: VintageGradleBuild
+    val currentBuild: BuildState
```

Comment:
💭  This is the main kind of change repeated in many places. The `BuildState` has everything the call sites need.